reference-implementation/web-platform-tests/readable-streams/bad-underlying-sources.js

```diff
@@ -0,0 +1,379 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+}
+
+test(function() {
+    var theError = new Error('a unique string');
+
+    assert_throws(theError, function() {
+        new ReadableStream({
+            get start() {
+                throw theError;
+            }
+        });
+    }, 'constructing the stream should re-throw the error');
+}, 'Underlying source start: throwing getter');
+
+test(function() {
+    var theError = new Error('a unique string');
+
+    assert_throws(theError, function() {
+        new ReadableStream({
+            start: function() {
+                throw theError;
+            }
+        });
+    }, 'constructing the stream should re-throw the error');
+}, 'Underlying source start: throwing method');
+
+var test1 = async_test('Underlying source: throwing pull getter (initial pull)');
+test1.step(function() {
+    var theError = new Error('a unique string');
+    var rs = new ReadableStream({
+        get pull() {
+            throw theError;
+        }
+    });
+
+    rs.getReader().closed.then(
+        test1.step_func(function() { assert_unreached('closed should not fulfill'); }),
+        test1.step_func(function(r) {
+            assert_equals(r, theError, 'closed should reject with the thrown error');
+            test1.done();
+        }));
+});
+
+var test2 = async_test('Underlying source: throwing pull method (initial pull)');
+test2.step(function() {
+    var theError = new Error('a unique string');
+    var rs = new ReadableStream({
+        pull: function() {
+            throw theError;
+        }
+    });
+
+    rs.getReader().closed.then(
+        test2.step_func(function() { assert_unreached('closed should not fulfill'); }),
+        test2.step_func(function(r) {
+            assert_equals(r, theError, 'closed should reject with the thrown error');
+            test2.done();
+        }));
+});
+
+var test3 = async_test('Underlying source: throwing pull getter (second pull)');
+test3.step(function() {
+    var theError = new Error('a unique string');
+    var counter = 0;
+    var rs = new ReadableStream({
+        get pull() {
+            ++counter;
+            if (counter === 1) {
+                return test3.step_func(function(c) { c.enqueue('a'); })
+            }
+
+            throw theError;
+        }
+    });
+    var reader = rs.getReader();
+
+    reader.read().then(test3.step_func(function(r) {
+        assert_object_equals(r, { value: 'a', done: false }, 'the chunk read should be correct');
+    }));
+
+    reader.closed.then(
+        test3.step_func(function() { assert_unreached('closed should not fulfill'); }),
+        test3.step_func(function(r) {
+            assert_equals(r, theError, 'closed should reject with the thrown error');
+            test3.done();
+        }));
+});
+
+var test4 = async_test('Underlying source: throwing pull method (second pull)');
```

Comment:
nitpick: you've changed the description to:
  Underlying source <method>: throwing method/getter
for start but not for pull.